test/hotspot/jtreg/TEST.ROOT

```diff
@@ -75,7 +75,6 @@ requires.properties= \
     vm.compiler2.enabled \
     vm.musl \
     docker.support \
-    test.vm.gc.nvdimm \
```

Comment:
`test/jtreg-ext/requires/VMProps.java` still references `test.vm.gc.nvdimm`, btw.

Reply:
The original change introduced the `_hrm` -> `*_hrm` change, so this is a straightforward reversal, no optimization :)

There has been one more removal possible by the `test.vm.gc.nvdimm` line in VMProps.java.

Thanks for your review!